### DISCUSSION TITLE
Question about Best Practice for Recording External Events

### DISCUSSION THREAD

**Discussion by du8die** [twentyhq/twenty#8948] [Unsolvable issues]
Hi All. I've been using Twenty since I think last week Wednesday. Over the past 5 years, we've bounced around between HubSpot, Nutshell, Zoho - and a host of other CRM products. There have ultimately been significant issues that hinder our use of each - and the pricing of some of them is simply insane. So far - Twenty meets all of our organizational needs - simply. That is a huge factor - because I want my sales team selling - not organizing. 

But, Twenty does not have bulk email and sequences. Not a problem. We use Mautic to do all of our sequences. Our deliverability is through the roof - since I get to provide my own smtp provider - and am not locked into whatever crap hubspot uses. 

So - over the past few days, I've used n8n to create a workflow that responds to all of a person record's events - and make sure that Mautic is in sync. I'm not syncing companies - because I don't need that for Mautic (at the moment). I have a custom object in Twenty that stores the name and Mautic ID's for the sequences and campaigns in Mautic. When you add a person to the Custom Object in Twenty - a webhook adds it to Mautic. Vice versa when you remove.

It's great. But, I want to sync email sends, opens and clicks back to Twenty. I'm not sure I want to use notes - because really this is nothing more than a "timeline event." I've seen the API for the timeline events - but it seems to me that you can't just create a timeline event. You must create something that gets attached to the event - like a note - which I don't really want to use.

So - is there any mechanism for creating a custom timeline event? What's the best practice for doing what I'm hoping to do here?

**Comment by aflatter** [comment]
Same here. I would like to archive Twenty's timeline as a trail of activity that happened in the context of a given opportunity. The idea is to use that for future analysis.

**Reply by aflatter** [reply]
I tried using the API to create a custom timeline activity but that just results in entries that don't have a description/caption. 
<img width="910" alt="grafik" src="https://github.com/user-attachments/assets/bf2b65be-0620-4aa9-8fdb-2f9883bd017b" />

**Comment by ianjohnsa** [comment]
I have started piloting twenty CRM for my own use recently and would also love to be able to add custom events to the timeline. In my case, I'm trying to record text/SMS sends or phone calls which seem like important events to record in a CRM but there doesn't appear to be a way to do that and have it show up in the timeline unless I'm missing something. I tried to create a custom object called 'activity log' and then record the events with a specific field to specify what they are (SMS, Call, Direct Mailing etc) but it doesn't show in the main timeline of course and even in the custom object section of the left pane for a specific person they aren't shown with a timestamp or anything so it's really not ideal. I am coming from Copper CRM and Copper handled this without issue with their timeline, including with custom events.
